gfx.cpp

```diff
@@ -2621,7 +2622,8 @@ class GfxInternal
         Program &gfx_program = programs_.insert(program);
         gfx_program.shader_model_ = shader_model;
         gfx_program.file_name_ = file_name;
-        gfx_program.file_path_ = file_path;
+        std::string const absolute_path = absolute(std::filesystem::path(file_path)).string();
+        gfx_program.file_path_ = absolute_path.c_str();
```

Comment:
Shader paths need to be absolute in order for the path detection to correctly work